### PR TITLE
feat: Add order margin end-point

### PR DESCRIPTION
**Pull request by ranjanrak**
Add the order margin end-point. Some internal refactor on **request** function to allow JSON content type POST request.

### CODE REVIEW

**Comment by vividvilla** [comment]
Setting request content type and serialization should be more generic. For example here you have added a flag called `postJSON` but in future there could be JSON put also. So for now you can rename `postJSON` to `is_json` or just `json`.

File: lib/connect.js [hunk 1]
```diff
@@ -158,10 +159,6 @@ var KiteConnect = function(params) {
         }
```
Comment (vividvilla) [inline lib/connect.js:159]: `paramsSerializer` should be removed since now you do serialization based on flag sent.

File: lib/connect.js [hunk 2]
```diff
@@ -666,6 +663,26 @@ regular).
...
+     * @param {string} params.transaction_type   eg. BUY, SELL  
+     * @param {string} params.variety   Order variety (regular, amo, bo, co etc.)
+     * @param {string} params.product   Margin product to use for the order
+     * @param {string} params.order_type   Order type (MARKET, LIMIT etc.)
+     * @param {number} params.quantity   Quantity of the order
+     * @param {number} params.price   Price at which the order is going to be placed (LIMIT orders)
+     * @param {number} params.trigger_price   Trigger price (for SL, SL-M, CO orders)
+     */
+    self.orderMargins = function(params){
+        return _post("order.margins", params, ...Array(2), true);
```
Comment (vividvilla) [inline lib/connect.js:683]: I didn't get what `...Array(2)` does here? may be I am missing something.

File: lib/connect.js [hunk 3]
```diff
@@ -1251,23 +1268,23 @@ regular).
         return data;
     }
 
-    function _get(route, params, responseType, responseTransformer) {
-        return request(route, "GET", params || {}, responseType, responseTransformer);
+    function _get(route, params, responseType, responseTransformer, postJson = false) {
```
Comment (vividvilla) [inline lib/connect.js:1271]: change flag `postJson` to something generic like `isJSON` or just `json`.

File: lib/connect.js [hunk 4]
```diff
@@ -1287,14 +1304,20 @@ regular).
         if (method === "GET" || method === "DELETE") {
             queryParams = params;
         } else {
-            payload = params;
+            if (postJson){
+                // post JSON payload
+                payload = JSON.stringify(params);
+            } else{
+            // post url encoded payload    
+            payload = querystring.stringify(params);
```
Comment (vividvilla) [inline lib/connect.js:1312]: Need a better alignment here.

**Review by vividvilla — CHANGES_REQUESTED** [review]

File: lib/connect.js (hunk 1 above)
Comment (ranjanrak) [reply lib/connect.js]: Done.

File: lib/connect.js (hunk 2 above)
Comment (ranjanrak) [reply lib/connect.js:680]: ...Array(2) sends blank/null value to **responseType** and **responseTransformer** parameter of **_post** function. In order to send true flag for optional json parameter.

File: lib/connect.js (hunk 3 above)
Comment (ranjanrak) [reply lib/connect.js]: Done. Changed it to **isJSON**.

File: lib/connect.js (hunk 4 above)
Comment (ranjanrak) [reply lib/connect.js:1309]: Done

**Review by ranjanrak — COMMENTED** [review]
@vividvilla all the above-requested changes are done. Please review the latest push.

File: lib/connect.js (hunk 2 above)
Comment (vividvilla) [reply lib/connect.js:680]: you can send `null` as well I guess.

**Review by vividvilla — APPROVED** [review]

**Pull request merged**